src/map.cpp

```diff
@@ -1083,12 +1083,15 @@ bool Map::isBlockOccluded(MapBlock *block, v3s16 cam_pos_nodes)
 
 	v3s16 pos_blockcenter = block->getPosRelative() + (MAP_BLOCKSIZE / 2);
 
-	// Starting step size, value between 1m and sqrt(3)m
-	float step = BS * 1.2f;
+	// Starting step size, slightly > sqrt(3),
+	// so that each block is hit at most once.
+	float step = BS * 1.7321f;
 	// Multiply step by each iteraction by 'stepfac' to reduce checks in distance
 	float stepfac = 1.05f;
 
-	float start_offset = BS * 1.0f;
+	// Take the camera's front view plane into account, and start slighly back.
+	// (TODO: use near_plane config)
+	float start_offset = -BS * 0.1f;
```

Comment:
This is something I can help with. This should be the distance of the camera (the point) from the viewport (the screen). Is there a good way to calculate that?

Reply:
Isn't that the value returned by [`ICameraSceneNode::getNearValue()`](http://irrlicht.sourceforge.net/docu/classirr_1_1scene_1_1_i_camera_scene_node.html#aee5dfccee2ec0b0cbcdb1079a6430a25)?

Wuzzy once posted this illustration:
![](https://i.stack.imgur.com/W4GSS.jpg)

Reply:
Thanks @sfan5 . I dug out my trusted old "Computer Graphics - Principles and Practice" book (I love it when I get a reason to do that).
Looks like the near clipping place actually has nothing do to the projection pane, it's just there to clip near objects to avoid distortions.

Somewhere is the actual projection pane (i.e. your screen). If I can easily find its coordinates I can place the screen in world coordinates and then use rays from its corners.

In any case. It turns out that part of the change is not actually needed, I'll remove that soon.